docs/web-apps/automated-testing/selenium.md

```diff
@@ -8,308 +8,307 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
+  such as `//input[@placeholder="Username"]`  
 
-You can use any of the WebDriver API **locator methods** to form locator expressions that find an element based on a specified locator type and value.  The Java `By` class, for example, could be used to contruct an expression that looks for an element by it's name value. `By.name("password")` looks for an element where `name` = `password`.
+:::tip Identifying Elements in HTML
 
-#### Finder Methods and the `WebElement` Interface
+In your Selenium test scripts, identify test elements by their `name` or `id` attribute value, since those are often the 
+simplest way to uniquely identify the element.
+:::
 
-Pass your By class expression as an argument of a WebDriver API **finder method** -- `findElement` (singular) and `findElements` (plural) -- to invoke it. For example, `findElement(By.name("password"))`.  
+#### Locator Methods
 
-Both finder methods search the DOM (Document Object Model) of the web page for elements that match the locator expression. The `findElement` returns the first matching element and stops, while `findElements` searches the entire page and returns a list of all matching elements.
+You can use any of the WebDriver API **locator methods** to form locator expressions that find an element based on a 
+specified locator type and value. In Java and .NET, locators are managed with a `By` class instance - `By.id("user-name")`. 
+In Python, the locator method is merged with the finder method (as described below) - `find_element_by_id("user-name")`. 
+Whereas Ruby uses key value pairs, typically as Hash values: `{id: "user-name"}`
 
-The following example invokes `findElement` on our `driver` instance to locate the username and password text input elements, and (optionally) the id attribute to locate the form element.
+Most of the elements in our Swag Labs example have multiple unique attributes that make it easy to
+identify them. For this example we can identify them in Java as follows:
 
+```java reference
+https://github.com/saucelabs-training/demo-java/blob/0105cf9/selenium-examples/src/test/java/com/saucedemo/login/W3CDemoTest.java#L47-L49
 ```
-import org.openqa.selenium.By;  
-import org.openqa.selenium.WebElement;  
-WebElement usernameElement = driver.findElement(By.name("username"));  
-WebElement passwordElement = driver.findElement(By.name("password"));
-WebElement formElement = driver.findElement(By.id("loginForm"));
-```
-
-### Step 4: Perform Actions on Located Elements
 
-Invoke an interaction method on an instance of the WebElement interface to simulate the user's interaction with the website elements you have located. Some basic interaction methods include:
+#### Finder Methods
 
-* The `sendKeys` method to enter text
-* The `clear` method to clear entered text
-* The `submit` method to submit a form  
+To find an element, pass your locator method as an argument of a WebDriver API **finder method**. The find element
```

Comment:
```suggestion
To find an element, pass your locator method as an argument of a WebDriver API **[finder method](https://www.selenium.dev/documentation/en/webdriver/web_element/)**. The find element
```

Reply:
Since there is more on that page than just finding elements, let's link to https://www.selenium.dev/documentation/en/webdriver/web_element/#find-element